### PR TITLE
Fix year jump

### LINKED ISSUE #761: Year jumps unexpectedly after guitar pick drag
Steps to reproduce: 
Start app, click on down arrow for the Year, the timeline does not update/change/move to correct year and the imagery does not update either.

### PR DESCRIPTION
**Pull request by Benjaki2**
Fixes #761  .

Changes in this pull request:

- [x] Combine various reverse/forward functions into one `animateByIncrement` function
- [x] Isolate timeouts from one global timeout for all date-change operations to two more specific timeouts.

@nasa-gibs/worldview

### CODE REVIEW

**Comment by Benjaki2** [comment]
@minniewong Could you test this locally and see if the date-Jump issue is fixed?

File: web/js/date/timeline-input.js [hunk 1]
```diff
@@ -10,177 +11,297 @@ import lodashParseInt from 'lodash/parseInt';
...
-      self.stop();
-    }
-  };
+    $(document)
+      .mouseout(stopper)
+      .keydown(function (event) {
+        if (event.target.nodeName === 'INPUT' || keyDown === event.keyCode) return;
+        switch (event.keyCode) {
+          case util.key.LEFT:
+            if (models.date.selectedZoom === 4) {
```
Comment (ZachTRice) [inline web/js/date/timeline-input.js]: This section of the code looks like it needs some latest changes from develop. There was an addition added to v2.3.0 which changed the left / right arrow functionality to reflect the current zoom level.
```
switch (models.date.selectedZoom) {
      case 1:
        animateReverse('year', -1);
        break;
      case 2:
        animateReverse('month', -1);
        break;
      case 3:
        animateReverse('day', -1);
        break;
      case 4:
        animateReverse('minute', -10);
        break;
}
```

File: web/js/date/timeline-input.js [hunk 2]
```diff
@@ -10,177 +11,297 @@ import lodashParseInt from 'lodash/parseInt';
...
+              animateByIncrement(-1, 'hour');
+            } else {
+              animateByIncrement(-1, 'day');
+            }
+            event.preventDefault();
+            break;
+          case util.key.RIGHT:
```
Comment (ZachTRice) [inline web/js/date/timeline-input.js]: Same as the above except with `animateForward`

File: web/js/date/timeline-input.js [hunk 3]
```diff
@@ -283,10 +407,10 @@ export function timelineInput(models, config, ui) {
       if ((selectedDateObj > tl.data.start()) &&
         (selectedDateObj <= util.today())) {
         var parent = selected.parent();
-        var sib = parent.next('div.input-wrapper.selectable')
```
Comment (ZachTRice) [inline web/js/date/timeline-input.js]: I am not able to tab from the day field back around to the year field. This is due to removing the `selectable` class logic here on this line (and the line below) + the toggleSubdaily function is no longer applying those classes (here: https://github.com/nasa-gibs/worldview/blob/fix-year-jump-3/web/js/layers/active.js#L428-L444)

File: web/js/date/timeline-input.js (hunk 1 above)
Comment (Benjaki2) [reply web/js/date/timeline-input.js]: Good find!

**Comment by Benjaki2** [comment]
@ZachTRice Thanks for the feedback. The requested changes are up. Could you take a again?

**Comment by ZachTRice** [comment]
@Benjaki2 I am still not able to tab correctly on the day time input (it should bring me back to the year input). Also, using the left / right arrow keys on the keyboard while in month / year zoom levels does not increment by month and year. It still increments by 1 day.  *Note, minute increments by 1 hour when I use the left and right arrow which is correct.

**Pull request merged**